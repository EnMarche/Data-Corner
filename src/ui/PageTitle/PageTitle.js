import { Grid, makeStyles } from '@material-ui/core'
import PropTypes from 'prop-types'

const useStyles = makeStyles(theme => ({
  pageTitle: {
    fontSize: '24px',
    fontWeight: '400',
    color: theme.palette.gray800,
    marginBottom: theme.spacing(2),
  },
}))

const PageTitle = ({ page, xs }) => {
  const classes = useStyles()

  return (
    <Grid item xs={xs} className={classes.pageTitle}>
      {page}
    </Grid>
  )
}

export default PageTitle

PageTitle.propTypes = {
  page: PropTypes.string.isRequired,
  xs: PropTypes.bool,
}
