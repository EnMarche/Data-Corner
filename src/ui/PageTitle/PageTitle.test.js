import { render } from '@testing-library/react'
import PageTitle from './PageTitle'

jest.mock('@material-ui/core', () => ({
  Grid: ({ children, breakpoints }) => <div xs={breakpoints}>{children}</div>,
  makeStyles: () => () => ({ title: 'title' }),
}))

describe('PageTitle', () => {
  it('displays a title', () => {
    const { container } = render(<PageTitle title="Title for test" xs={0} />)

    expect(container).toMatchSnapshot()
  })
})
