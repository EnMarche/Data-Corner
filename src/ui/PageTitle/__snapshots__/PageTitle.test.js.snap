// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTitle displays a title 1`] = `
<div>
  <div
    class="title"
  >
    Title for test
  </div>
</div>
`;
